source/SIL.AppBuilder.Portal/common/databaseProxy/OrganizationMemberships.ts

```diff
@@ -0,0 +1,83 @@
+import prisma from '../prisma.js';
+
+export async function acceptOrganizationInvite(userId: number, inviteToken: string) {
+  const invite = await prisma.organizationMembershipInvites.findFirst({
+    where: {
+      Token: inviteToken
+    }
+  });
+  if (!invite || invite.Redeemed || !invite.Expires || invite.Expires < new Date()) return false;
+  // Check if the user is already a member of the organization
+  // This could be done such that they can accept the invite to get the roles and groups
+  const existingMembership = await prisma.organizationMemberships.findFirst({
+    where: {
+      UserId: userId,
+      OrganizationId: invite.OrganizationId
+    }
+  });
+  if (existingMembership) return false;
+  await prisma.organizationMemberships.create({
+    data: {
+      UserId: userId,
+      OrganizationId: invite.OrganizationId
+    }
+  });
+  await prisma.userRoles.createMany({
+    data: invite.Roles.split(',')
```

Comment:
Remember to change this if you do follow my suggestion regarding the array field type